### PR TITLE
Uses unmodifiableList for transactions in Block.getTransactionsList

### PR DESCRIPTION
**Pull request by tinchou**
This adds some basic immutability checks. It uncovers some tech debt in the bridge tests.

### CODE REVIEW

**Comment by tinchou** [comment]
Also, removes `CopyOnWrite` usage, which would be inefficient if anyone ends up using it.

File: rskj-core/src/main/java/org/ethereum/core/Block.java [hunk 1]
```diff
@@ -416,6 +421,8 @@ public String toFlatString() {
     }
 
     private void parseTxs(RLPList txTransactions) {
+        // TODO(mc) check if we can avoid copying existing transactions
+        ArrayList<Transaction> builder = new ArrayList<>(this.transactionsList);
```
Comment (ajlopez) [inline rskj-core/src/main/java/org/ethereum/core/Block.java]: Reconsider use case: the transaction list is expected to be empty

**Comment by tinchou** [comment]
ping @ajlopez

**Comment by tinchou** [comment]
@diega would you like to review?

File: rskj-core/src/main/java/org/ethereum/core/Block.java [hunk 2]
```diff
@@ -455,6 +458,8 @@ public String toFlatString() {
     }
 
     private void parseTxs(RLPList txTransactions) {
+        ArrayList<Transaction> builder = new ArrayList<>();
```
Comment (diega) [inline rskj-core/src/main/java/org/ethereum/core/Block.java]: It's recommended to use an interface whenever you can, in this case _builder_ could be  `List<Transaction>`. Also I think the name of the variable could be improved, what about _parsedTransactions_?

File: rskj-core/networkStateExporterTest.json [hunk 3]
```diff
@@ -1,14 +0,0 @@
-{
```
Comment (diega) [inline rskj-core/networkStateExporterTest.json]: what's this removal?

Comment (tinchou) [reply rskj-core/networkStateExporterTest.json]: Thanks for spotting!

**Pull request merged**